### ISSUE TITLE
Problem connecting to bitvise WinSSHD 7.42 server

### ISSUE THREAD

**Issue by ttraskback** [phpseclib/phpseclib#1448]
I have tried both 1.x and 2.x version of this package and both result in same problem.

I can conenct with he commandline sftp client.

Code:
```
<?php 
include 'vendor/autoload.php';
define('NET_SSH2_LOGGING', 3);

$sftp = new \phpseclib\Net\SFTP('ip');
if (!$sftp->login('user', 'passwd')) {
  echo "Log: \n";
  var_dump($sftp->getLog());
  echo "\n Errors: \n";
  var_dump($sftp->getErrors());
  exit('Login Failed');
}
```

Log:
```
<-
00000000  53:53:48:2d:32:2e:30:2d:37:2e:34:32:20:46:6c:6f  SSH-2.0-7.42 Flo
00000010  77:53:73:68:3a:20:42:69:74:76:69:73:65:20:53:53  wSsh: Bitvise SS
00000020  48:20:53:65:72:76:65:72:20:28:57:69:6e:53:53:48  H Server (WinSSH
00000030  44:29:20:37:2e:34:32:0d:0a                       D) 7.42..



->
00000000  53:53:48:2d:32:2e:30:2d:70:68:70:73:65:63:6c:69  SSH-2.0-phpsecli
00000010  62:5f:32:2e:30:20:28:6f:70:65:6e:73:73:6c:2c:20  b_2.0 (openssl, 
00000020  62:63:6d:61:74:68:29:0d:0a                       bcmath)..



-> NET_SSH2_MSG_KEXINIT (since last: 0.1123, network: 0.0002s)
00000000  fd:2e:0b:fb:ed:37:81:cb:61:fa:08:85:cd:0c:b1:ae  .....7..a.......
00000010  00:00:00:7e:64:69:66:66:69:65:2d:68:65:6c:6c:6d  ...~diffie-hellm
00000020  61:6e:2d:67:72:6f:75:70:2d:65:78:63:68:61:6e:67  an-group-exchang
00000030  65:2d:73:68:61:32:35:36:2c:64:69:66:66:69:65:2d  e-sha256,diffie-
00000040  68:65:6c:6c:6d:61:6e:2d:67:72:6f:75:70:2d:65:78  hellman-group-ex
00000050  63:68:61:6e:67:65:2d:73:68:61:31:2c:64:69:66:66  change-sha1,diff
00000060  69:65:2d:68:65:6c:6c:6d:61:6e:2d:67:72:6f:75:70  ie-hellman-group
00000070  31:34:2d:73:68:61:31:2c:64:69:66:66:69:65:2d:68  14-sha1,diffie-h
00000080  65:6c:6c:6d:61:6e:2d:67:72:6f:75:70:31:2d:73:68  ellman-group1-sh
00000090  61:31:00:00:00:29:72:73:61:2d:73:68:61:32:2d:32  a1...)rsa-sha2-2
000000a0  35:36:2c:72:73:61:2d:73:68:61:32:2d:35:31:32:2c  56,rsa-sha2-512,
000000b0  73:73:68:2d:72:73:61:2c:73:73:68:2d:64:73:73:00  ssh-rsa,ssh-dss.
000000c0  00:00:ff:61:65:73:31:32:38:2d:63:74:72:2c:61:65  ...aes128-ctr,ae
000000d0  73:31:39:32:2d:63:74:72:2c:61:65:73:32:35:36:2d  s192-ctr,aes256-
000000e0  63:74:72:2c:61:65:73:31:32:38:2d:63:62:63:2c:61  ctr,aes128-cbc,a
000000f0  65:73:31:39:32:2d:63:62:63:2c:61:65:73:32:35:36  es192-cbc,aes256
00000100  2d:63:62:63:2c:62:6c:6f:77:66:69:73:68:2d:63:74  -cbc,blowfish-ct
00000110  72:2c:62:6c:6f:77:66:69:73:68:2d:63:62:63:2c:33  r,blowfish-cbc,3
00000120  64:65:73:2d:63:74:72:2c:33:64:65:73:2d:63:62:63  des-ctr,3des-cbc
00000130  2c:74:77:6f:66:69:73:68:31:32:38:2d:63:74:72:2c  ,twofish128-ctr,
00000140  74:77:6f:66:69:73:68:31:39:32:2d:63:74:72:2c:74  twofish192-ctr,t
00000150  77:6f:66:69:73:68:32:35:36:2d:63:74:72:2c:74:77  wofish256-ctr,tw
00000160  6f:66:69:73:68:31:32:38:2d:63:62:63:2c:74:77:6f  ofish128-cbc,two
00000170  66:69:73:68:31:39:32:2d:63:62:63:2c:74:77:6f:66  fish192-cbc,twof
00000180  69:73:68:32:35:36:2d:63:62:63:2c:74:77:6f:66:69  ish256-cbc,twofi
00000190  73:68:2d:63:62:63:2c:61:72:63:66:6f:75:72:32:35  sh-cbc,arcfour25
000001a0  36:2c:61:72:63:66:6f:75:72:32:35:36:2c:61:72:63  6,arcfour256,arc
000001b0  66:6f:75:72:31:32:38:2c:61:72:63:66:6f:75:72:31  four128,arcfour1
000001c0  32:38:00:00:00:ff:61:65:73:31:32:38:2d:63:74:72  28....aes128-ctr
000001d0  2c:61:65:73:31:39:32:2d:63:74:72:2c:61:65:73:32  ,aes192-ctr,aes2
000001e0  35:36:2d:63:74:72:2c:61:65:73:31:32:38:2d:63:62  56-ctr,aes128-cb
000001f0  63:2c:61:65:73:31:39:32:2d:63:62:63:2c:61:65:73  c,aes192-cbc,aes
00000200  32:35:36:2d:63:62:63:2c:62:6c:6f:77:66:69:73:68  256-cbc,blowfish
00000210  2d:63:74:72:2c:62:6c:6f:77:66:69:73:68:2d:63:62  -ctr,blowfish-cb
00000220  63:2c:33:64:65:73:2d:63:74:72:2c:33:64:65:73:2d  c,3des-ctr,3des-
00000230  63:62:63:2c:74:77:6f:66:69:73:68:31:32:38:2d:63  cbc,twofish128-c
00000240  74:72:2c:74:77:6f:66:69:73:68:31:39:32:2d:63:74  tr,twofish192-ct
00000250  72:2c:74:77:6f:66:69:73:68:32:35:36:2d:63:74:72  r,twofish256-ctr
00000260  2c:74:77:6f:66:69:73:68:31:32:38:2d:63:62:63:2c  ,twofish128-cbc,
00000270  74:77:6f:66:69:73:68:31:39:32:2d:63:62:63:2c:74  twofish192-cbc,t
00000280  77:6f:66:69:73:68:32:35:36:2d:63:62:63:2c:74:77  wofish256-cbc,tw
00000290  6f:66:69:73:68:2d:63:62:63:2c:61:72:63:66:6f:75  ofish-cbc,arcfou
000002a0  72:32:35:36:2c:61:72:63:66:6f:75:72:32:35:36:2c  r256,arcfour256,
000002b0  61:72:63:66:6f:75:72:31:32:38:2c:61:72:63:66:6f  arcfour128,arcfo
000002c0  75:72:31:32:38:00:00:00:39:68:6d:61:63:2d:73:68  ur128...9hmac-sh
000002d0  61:32:2d:32:35:36:2c:68:6d:61:63:2d:73:68:61:31  a2-256,hmac-sha1
000002e0  2d:39:36:2c:68:6d:61:63:2d:73:68:61:31:2c:68:6d  -96,hmac-sha1,hm
000002f0  61:63:2d:6d:64:35:2d:39:36:2c:68:6d:61:63:2d:6d  ac-md5-96,hmac-m
00000300  64:35:00:00:00:39:68:6d:61:63:2d:73:68:61:32:2d  d5...9hmac-sha2-
00000310  32:35:36:2c:68:6d:61:63:2d:73:68:61:31:2d:39:36  256,hmac-sha1-96
00000320  2c:68:6d:61:63:2d:73:68:61:31:2c:68:6d:61:63:2d  ,hmac-sha1,hmac-
00000330  6d:64:35:2d:39:36:2c:68:6d:61:63:2d:6d:64:35:00  md5-96,hmac-md5.
00000340  00:00:04:6e:6f:6e:65:00:00:00:04:6e:6f:6e:65:00  ...none....none.
00000350  00:00:00:00:00:00:00:00:00:00:00:00              ............



<- NET_SSH2_MSG_KEXINIT (since last: 0.0104, network: 0s)
00000000  14:8f:aa:84:c6:d1:67:27:9b:07:81:99:0a:43:7c:e4  ......g'.....C|.
00000010  00:00:01:d1:63:75:72:76:65:32:35:35:31:39:2d:73  ....curve25519-s
00000020  68:61:32:35:36:2c:63:75:72:76:65:32:35:35:31:39  ha256,curve25519
00000030  2d:73:68:61:32:35:36:40:6c:69:62:73:73:68:2e:6f  -sha256@libssh.o
00000040  72:67:2c:65:63:64:68:2d:73:68:61:32:2d:31:2e:33  rg,ecdh-sha2-1.3
00000050  2e:31:33:32:2e:30:2e:31:30:2c:65:63:64:68:2d:73  .132.0.10,ecdh-s
00000060  68:61:32:2d:6e:69:73:74:70:32:35:36:2c:65:63:64  ha2-nistp256,ecd
00000070  68:2d:73:68:61:32:2d:6e:69:73:74:70:33:38:34:2c  h-sha2-nistp384,
00000080  65:63:64:68:2d:73:68:61:32:2d:6e:69:73:74:70:35  ecdh-sha2-nistp5
00000090  32:31:2c:64:69:66:66:69:65:2d:68:65:6c:6c:6d:61  21,diffie-hellma
000000a0  6e:2d:67:72:6f:75:70:2d:65:78:63:68:61:6e:67:65  n-group-exchange
000000b0  2d:73:68:61:32:35:36:2c:64:69:66:66:69:65:2d:68  -sha256,diffie-h
000000c0  65:6c:6c:6d:61:6e:2d:67:72:6f:75:70:2d:65:78:63  ellman-group-exc
000000d0  68:61:6e:67:65:2d:73:68:61:31:2c:64:69:66:66:69  hange-sha1,diffi
000000e0  65:2d:68:65:6c:6c:6d:61:6e:2d:67:72:6f:75:70:31  e-hellman-group1
000000f0  35:2d:73:68:61:35:31:32:2c:64:69:66:66:69:65:2d  5-sha512,diffie-
00000100  68:65:6c:6c:6d:61:6e:2d:67:72:6f:75:70:31:34:2d  hellman-group14-
00000110  73:68:61:32:35:36:2c:64:69:66:66:69:65:2d:68:65  sha256,diffie-he
00000120  6c:6c:6d:61:6e:2d:67:72:6f:75:70:31:34:2d:73:68  llman-group14-sh
00000130  61:31:2c:67:73:73:2d:67:65:78:2d:73:68:61:31:2d  a1,gss-gex-sha1-
00000140  74:6f:57:4d:35:53:6c:77:35:45:77:38:4d:71:6b:61  toWM5Slw5Ew8Mqka
00000150  79:2b:61:6c:32:67:3d:3d:2c:67:73:73:2d:67:72:6f  y+al2g==,gss-gro
00000160  75:70:31:35:2d:73:68:61:35:31:32:2d:74:6f:57:4d  up15-sha512-toWM
00000170  35:53:6c:77:35:45:77:38:4d:71:6b:61:79:2b:61:6c  5Slw5Ew8Mqkay+al
00000180  32:67:3d:3d:2c:67:73:73:2d:67:72:6f:75:70:31:34  2g==,gss-group14
00000190  2d:73:68:61:32:35:36:2d:74:6f:57:4d:35:53:6c:77  -sha256-toWM5Slw
000001a0  35:45:77:38:4d:71:6b:61:79:2b:61:6c:32:67:3d:3d  5Ew8Mqkay+al2g==
000001b0  2c:67:73:73:2d:67:72:6f:75:70:31:34:2d:73:68:61  ,gss-group14-sha
000001c0  31:2d:74:6f:57:4d:35:53:6c:77:35:45:77:38:4d:71  1-toWM5Slw5Ew8Mq
000001d0  6b:61:79:2b:61:6c:32:67:3d:3d:2c:65:78:74:2d:69  kay+al2g==,ext-i
000001e0  6e:66:6f:2d:73:00:00:00:35:72:73:61:2d:73:68:61  nfo-s...5rsa-sha
000001f0  32:2d:35:31:32:2c:72:73:61:2d:73:68:61:32:2d:32  2-512,rsa-sha2-2
00000200  35:36:2c:73:73:68:2d:72:73:61:2c:65:63:64:73:61  56,ssh-rsa,ecdsa
00000210  2d:73:68:61:32:2d:6e:69:73:74:70:33:38:34:00:00  -sha2-nistp384..
00000220  00:57:61:65:73:32:35:36:2d:67:63:6d:40:6f:70:65  .Waes256-gcm@ope
00000230  6e:73:73:68:2e:63:6f:6d:2c:61:65:73:31:32:38:2d  nssh.com,aes128-
00000240  67:63:6d:40:6f:70:65:6e:73:73:68:2e:63:6f:6d:2c  gcm@openssh.com,
00000250  61:65:73:32:35:36:2d:63:74:72:2c:61:65:73:31:39  aes256-ctr,aes19
00000260  32:2d:63:74:72:2c:61:65:73:31:32:38:2d:63:74:72  2-ctr,aes128-ctr
00000270  2c:33:64:65:73:2d:63:74:72:00:00:00:57:61:65:73  ,3des-ctr...Waes
00000280  32:35:36:2d:67:63:6d:40:6f:70:65:6e:73:73:68:2e  256-gcm@openssh.
00000290  63:6f:6d:2c:61:65:73:31:32:38:2d:67:63:6d:40:6f  com,aes128-gcm@o
000002a0  70:65:6e:73:73:68:2e:63:6f:6d:2c:61:65:73:32:35  penssh.com,aes25
000002b0  36:2d:63:74:72:2c:61:65:73:31:39:32:2d:63:74:72  6-ctr,aes192-ctr
000002c0  2c:61:65:73:31:32:38:2d:63:74:72:2c:33:64:65:73  ,aes128-ctr,3des
000002d0  2d:63:74:72:00:00:00:17:68:6d:61:63:2d:73:68:61  -ctr....hmac-sha
000002e0  32:2d:32:35:36:2c:68:6d:61:63:2d:73:68:61:31:00  2-256,hmac-sha1.
000002f0  00:00:17:68:6d:61:63:2d:73:68:61:32:2d:32:35:36  ...hmac-sha2-256
00000300  2c:68:6d:61:63:2d:73:68:61:31:00:00:00:09:7a:6c  ,hmac-sha1....zl
00000310  69:62:2c:6e:6f:6e:65:00:00:00:09:7a:6c:69:62:2c  ib,none....zlib,
00000320  6e:6f:6e:65:00:00:00:00:00:00:00:00:00:00:00:00  none............
00000330  00                                               .



-> UNKNOWN (34) (since last: 0.0091, network: 0.0003s)
00000000  00:00:06:00:00:00:08:00:00:00:10:00              ............



<- NET_SSH2_MSG_KEXDH_REPLY (since last: 0.0433, network: 0.042s)
00000000  00:00:01:01:00:ff:ff:ff:ff:ff:ff:ff:ff:c9:0f:da  ................
00000010  a2:21:68:c2:34:c4:c6:62:8b:80:dc:1c:d1:29:02:4e  .!h.4..b.....).N
00000020  08:8a:67:cc:74:02:0b:be:a6:3b:13:9b:22:51:4a:08  ..g.t....;.."QJ.
00000030  79:8e:34:04:dd:ef:95:19:b3:cd:3a:43:1b:30:2b:0a  y.4.......:C.0+.
00000040  6d:f2:5f:14:37:4f:e1:35:6d:6d:51:c2:45:e4:85:b5  m._.7O.5mmQ.E...
00000050  76:62:5e:7e:c6:f4:4c:42:e9:a6:37:ed:6b:0b:ff:5c  vb^~..LB..7.k..\
00000060  b6:f4:06:b7:ed:ee:38:6b:fb:5a:89:9f:a5:ae:9f:24  ......8k.Z.....$
00000070  11:7c:4b:1f:e6:49:28:66:51:ec:e4:5b:3d:c2:00:7c  .|K..I(fQ..[=..|
00000080  b8:a1:63:bf:05:98:da:48:36:1c:55:d3:9a:69:16:3f  ..c....H6.U..i.?
00000090  a8:fd:24:cf:5f:83:65:5d:23:dc:a3:ad:96:1c:62:f3  ..$._.e]#.....b.
000000a0  56:20:85:52:bb:9e:d5:29:07:70:96:96:6d:67:0c:35  V .R...).p..mg.5
000000b0  4e:4a:bc:98:04:f1:74:6c:08:ca:18:21:7c:32:90:5e  NJ....tl...!|2.^
000000c0  46:2e:36:ce:3b:e3:9e:77:2c:18:0e:86:03:9b:27:83  F.6.;..w,.....'.
000000d0  a2:ec:07:a2:8f:b5:c5:5d:f0:6f:4c:52:c9:de:2b:cb  .......].oLR..+.
000000e0  f6:95:58:17:18:39:95:49:7c:ea:95:6a:e5:15:d2:26  ..X..9.I|..j...&
000000f0  18:98:fa:05:10:15:72:8e:5a:8a:ac:aa:68:ff:ff:ff  ......r.Z...h...
00000100  ff:ff:ff:ff:ff:00:00:00:01:02                    ..........



-> UNKNOWN (32) (since last: 0.0537, network: 0.0003s)
00000000  00:00:01:00:6a:28:26:e9:55:c0:3a:29:20:a5:6f:5a  ....j(&.U.:) .oZ
00000010  03:66:2a:a3:8a:6d:01:21:f6:83:66:39:97:ba:91:93  .f*..m.!..f9....
00000020  c2:69:0d:a5:e7:36:3c:d6:da:84:f2:b1:4d:74:5f:f8  .i...6......Mt_.
00000030  24:83:c0:b8:d6:0d:d8:10:9c:a8:8f:c3:08:1b:50:47  $.............PG
00000040  52:4f:32:75:50:49:aa:1b:9b:01:8e:92:d5:36:49:61  RO2uPI.......6Ia
00000050  04:77:50:30:bb:ca:a8:a1:51:83:38:5b:e4:2a:16:04  .wP0....Q.8[.*..
00000060  ba:80:63:34:bb:77:33:37:fd:5e:7b:b3:1a:57:21:58  ..c4.w37.^{..W!X
00000070  14:b1:60:85:82:05:7c:38:08:43:db:51:ca:6c:a2:2a  ..`...|8.C.Q.l.*
00000080  43:f7:b1:4e:a2:7d:73:d0:a2:49:bd:72:ea:2c:2b:75  C..N.}s..I.r.,+u
00000090  82:83:b2:b7:30:58:f3:e5:df:1c:73:92:10:bb:ba:5b  ....0X....s....[
000000a0  fa:d7:0f:24:d5:15:98:bc:bf:70:bc:66:8a:80:83:53  ...$.....p.f...S
000000b0  3f:dc:9f:4c:1d:89:de:38:72:60:4a:8f:f2:09:aa:bc  ?..L...8r`J.....
000000c0  aa:36:d6:f1:b1:8d:a5:79:9e:f9:98:dd:72:67:a8:9f  .6.....y....rg..
000000d0  8a:16:81:2e:3b:f5:b7:c3:d5:13:b0:d6:c8:b1:e7:87  ....;...........
000000e0  18:76:53:c4:4e:c8:53:4c:c7:7f:7e:57:c0:08:25:6b  .vS.N.SL..~W..%k
000000f0  dc:34:cf:cf:31:64:f3:5d:7d:3b:58:2e:ad:c7:23:a7  .4..1d.]};X...#.
00000100  58:1a:d9:b7                                      X...

```

And sftp verbouse output:
```
OpenSSH_7.2p2 Ubuntu-4ubuntu2.2, OpenSSL 1.0.2g  1 Mar 2016
debug1: Reading configuration data /home/vagrant/.ssh/config
debug1: /home/vagrant/.ssh/config line 1: Applying options for sshtest
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: Connecting to localhost [::1] port 2003.
debug1: Connection established.
debug1: identity file /home/vagrant/.ssh/id_rsa type 1
debug1: key_load_public: No such file or directory
debug1: identity file /home/vagrant/.ssh/id_rsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/vagrant/.ssh/id_dsa type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/vagrant/.ssh/id_dsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/vagrant/.ssh/id_ecdsa type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/vagrant/.ssh/id_ecdsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/vagrant/.ssh/id_ed25519 type -1
debug1: key_load_public: No such file or directory
debug1: identity file /home/vagrant/.ssh/id_ed25519-cert type -1
debug1: Enabling compatibility mode for protocol 2.0
debug1: Local version string SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.2
debug1: Remote protocol version 2.0, remote software version 7.42 FlowSsh: Bitvise SSH Server (WinSSHD) 7.42
debug1: no match: 7.42 FlowSsh: Bitvise SSH Server (WinSSHD) 7.42
debug1: Authenticating to localhost:2003 as 'akeneo'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha256@libssh.org
debug1: kex: host key algorithm: ecdsa-sha2-nistp384
debug1: kex: server->client cipher: aes128-ctr MAC: hmac-sha2-256 compression: none
debug1: kex: client->server cipher: aes128-ctr MAC: hmac-sha2-256 compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp384 SHA256:mHExqqyYZ8n46CuIWcR6hUECqTAZKVzW+cIscbpqD6c
debug1: Host '[localhost]:2003' is known and matches the ECDSA host key.
debug1: Found key in /home/vagrant/.ssh/known_hosts:9
debug1: rekey after 4294967296 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: rekey after 4294967296 blocks
debug1: SSH2_MSG_NEWKEYS received
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,ecdsa-sha2-1.3.132.0.10,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,rsa-sha2-512,rsa-sha2-256,ssh-rsa>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,gssapi-with-mic,password
debug1: Next authentication method: gssapi-with-mic
debug1: Unspecified GSS failure.  Minor code may provide more information
No Kerberos credentials available

debug1: Unspecified GSS failure.  Minor code may provide more information
No Kerberos credentials available

debug1: Unspecified GSS failure.  Minor code may provide more information


debug1: Unspecified GSS failure.  Minor code may provide more information
No Kerberos credentials available
```

**Comment by terrafrost** [comment]
Could you provide me the IP address to connect to the SSH server myself so that I might be able to reproduce the issue and test possible fixes? I wouldn't need any login credentials since it seems like the issue is with the key exchange process. If you'd be willing to do this you could email the info to terrafrost@php.net.

Failing that I can try to install bitvise WinSSHD 7.42 myself.

Thanks!

**Comment by ttraskback** [comment]
> Could you provide me the IP address to connect to the SSH server myself so that I might be able to reproduce the issue and test possible fixes? I wouldn't need any login credentials since it seems like the issue is with the key exchange process. If you'd be willing to do this you could email the info to [terrafrost@php.net](mailto:terrafrost@php.net).
> 
> Failing that I can try to install bitvise WinSSHD 7.42 myself.
> 
> Thanks!

Hi,
Unfortunately i can't do that as it's out of our control and it's behind a firewall that ip restricts access.

Do you think updating bitvise could help? I could try suggesting that to the receiving side if it might solve the issue.

**Comment by terrafrost** [comment]
> Do you think updating bitvise could help? I could try suggesting that to the receiving side if it might solve the issue.

I tried to connect to Bitvise SSH Server 8.39 and was able to do so without issue with phpseclib 2.0.23.

I think the issue may have something to do with this:

```
-> UNKNOWN (32) (since last: 0.0537, network: 0.0003s)
00000000  00:00:01:00:6a:28:26:e9:55:c0:3a:29:20:a5:6f:5a  ....j(&.U.:) .oZ
00000010  03:66:2a:a3:8a:6d:01:21:f6:83:66:39:97:ba:91:93  .f*..m.!..f9....
00000020  c2:69:0d:a5:e7:36:3c:d6:da:84:f2:b1:4d:74:5f:f8  .i...6......Mt_.
00000030  24:83:c0:b8:d6:0d:d8:10:9c:a8:8f:c3:08:1b:50:47  $.............PG
00000040  52:4f:32:75:50:49:aa:1b:9b:01:8e:92:d5:36:49:61  RO2uPI.......6Ia
00000050  04:77:50:30:bb:ca:a8:a1:51:83:38:5b:e4:2a:16:04  .wP0....Q.8[.*..
00000060  ba:80:63:34:bb:77:33:37:fd:5e:7b:b3:1a:57:21:58  ..c4.w37.^{..W!X
00000070  14:b1:60:85:82:05:7c:38:08:43:db:51:ca:6c:a2:2a  ..`...|8.C.Q.l.*
00000080  43:f7:b1:4e:a2:7d:73:d0:a2:49:bd:72:ea:2c:2b:75  C..N.}s..I.r.,+u
00000090  82:83:b2:b7:30:58:f3:e5:df:1c:73:92:10:bb:ba:5b  ....0X....s....[
000000a0  fa:d7:0f:24:d5:15:98:bc:bf:70:bc:66:8a:80:83:53  ...$.....p.f...S
000000b0  3f:dc:9f:4c:1d:89:de:38:72:60:4a:8f:f2:09:aa:bc  ?..L...8r`J.....
000000c0  aa:36:d6:f1:b1:8d:a5:79:9e:f9:98:dd:72:67:a8:9f  .6.....y....rg..
000000d0  8a:16:81:2e:3b:f5:b7:c3:d5:13:b0:d6:c8:b1:e7:87  ....;...........
000000e0  18:76:53:c4:4e:c8:53:4c:c7:7f:7e:57:c0:08:25:6b  .vS.N.SL..~W..%k
000000f0  dc:34:cf:cf:31:64:f3:5d:7d:3b:58:2e:ad:c7:23:a7  .4..1d.]};X...#.
00000100  58:1a:d9:b7                                      X...
```
That's NET_SSH2_MSG_KEXDH_GEX_INIT. The server should be responding with NET_SSH2_MSG_KEXDH_GEX_REPLY but it's not (the fact that the logs show UNKNOWN is actually a bug that was fixed with https://github.com/phpseclib/phpseclib/commit/500e3225a839825f92033f9909794c55a8d0536c#diff-82a5983a6daace4acbcc65edf2dfb605)

Anyway, I assume the logs following the "And sftp verbouse output" section are from different clients? Because the phpseclib logs seem to show a connection that's _not_ being made whereas the "sftp verbouse output" appears to show a successful connection (albeit with a login failure).

I mean, I will say that phpseclib doesn't support GSSAPI but that shouldn't be an issue...

**Comment by ttraskback** [comment]
Hi,
Thanks for investigating i will see if they can update the bitvise server and update ticket accordingly.

**Comment by ttraskback** [comment]
Hi,
Issue was fixed with updating to 8.39.
Thanks for quick support
